### PR TITLE
[CS-16] Backend authorization

### PR DESCRIPTION
**Pull request by twistzzmc**

### CODE REVIEW

File: rest/__main__.py [hunk 1]
```diff
@@ -17,4 +23,100 @@ def hello():
     return response
 
 
+@app.route('/user', methods=['POST'])
```
Comment (jmysliv) [inline rest/__main__.py:26]: Put all this routes in different file, we don't want to keep all routes in one file.

File: rest/__main__.py [hunk 2]
```diff
@@ -17,4 +23,100 @@ def hello():
...
+
+@app.route('/user/editorRequests', methods=['GET'])
+def admin_editor_requests():
+    if 'email' not in request.cookies:
+        return app.response_class(response=json.dumps({'cookie': False}), status=401, mimetype='application/json')
+
+    email = request.cookies.get('email')
+    user = User.query.filter_by(email=email).first()
+
+    if not user or not user.is_editor:
```
Comment (jmysliv) [inline rest/__main__.py:109]: I think only admin should have access to editor requests not all editors.

File: rest/__main__.py [hunk 3]
```diff
@@ -17,4 +23,100 @@ def hello():
...
+        return app.response_class(response=json.dumps({'is_editor': False}), status=403, mimetype='application/json')
+
+    users = User.query.all()
```
Comment (jmysliv) [inline rest/__main__.py:112]: This get method should returns only editor requests, not every user in our database.

**Review by jmysliv — CHANGES_REQUESTED** [review]
Fix pylint errors.

File: rest/routes/user.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+
+@user_url.route('/auth', methods=['POST'])
+def login():
+    email = request.form.get('email')
+    password = request.form.get('password')
+
+    user = User.query.filter_by(email=email).first()
+
+    if user and user.password == password:
```
[the remark this answers is not in the document]
Comment (twistzzmc) [reply rest/routes/user.py:69]: Pushed again before reading it, sorry. I'm gonna fix it

File: tests/test_user.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            'password': '12345', 'editorRequest': False}
+    assert requests.post(URL_USER, data=data).status_code == 400
+
+    data = {'name': '123', 'email': 'proper@test.agh.edu.pl',
+            'password': '45', 'editorRequest': False}
+    assert requests.post(URL_USER, data=data).status_code == 400
+
+    data = {'name': 'proper_name ', 'email': 'proper@test.agh.edu.pl',
+            'password': '12345', 'editorRequest': False}
+    assert requests.post(URL_USER, data=data).status_code == 400
```
Comment (twrdyyy) [inline tests/test_user.py:48]: put in separate file

File: tests/test_user.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    requests.post(URL_USER, data=data)
+    login_data = {'email': 'login@agh.edu.pl', 'password': '12345'}
+    assert requests.post(URL_USER_AUTH, data=login_data).status_code == 200
+
+    login_data = {'email': 'login2@agh.edu.pl', 'password': '12345'}
+    assert requests.post(URL_USER_AUTH, data=login_data).status_code == 400
+
+    login_data = {'email': 'login@agh.edu.pl', 'password': '123435'}
+    assert requests.post(URL_USER_AUTH, data=login_data).status_code == 400
+
```
Comment (twrdyyy) [inline tests/test_user.py:63]: same here

File: tests/test_user.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    response = requests.post(URL_USER_AUTH, data=login_data)
+    assert response.status_code == 200
+    data = json.loads(response.content.decode('utf-8'))
+    assert data['role'] == 'editor'
+
+    login_data = {'email': 'admin@admin.agh.edu.pl', 'password': 'admin'}
+    response = requests.post(URL_USER_AUTH, data=login_data)
+    assert response.status_code == 200
+    data = json.loads(response.content.decode('utf-8'))
+    assert data['role'] == 'admin'
```
Comment (twrdyyy) [inline tests/test_user.py:85]: and here

File: tests/test_user.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+    response = requests.get(
+        URL_USER_EDITOR,
+        cookies={'email': 'admin@admin.agh.edu.pl'}
+    )
+    assert response.status_code == 200
+    data = json.loads(response.content.decode('utf-8'))
+    assert len(data) == 1
+    assert data[0]['name'] == 'proper_name'
+    assert data[0]['user_email'] == 'register@agh.edu.pl'
```
Comment (twrdyyy) [inline tests/test_user.py:113]: here

File: tests/test_user.py [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        URL_USER_EDITOR,
+        cookies={'email': 'admin@admin.agh.edu.pl'}
+    )
+    assert response.status_code == 200
+    data = json.loads(response.content.decode('utf-8'))
+    assert len(data) == 0
+
+    data = {'name': 'proper_name', 'email': 'register@agh.edu.pl'}
+    response = requests.post(URL_USER_EDITOR, data=data)
+    assert response.status_code == 400
```
Comment (twrdyyy) [inline tests/test_user.py:133]: here as well :)

File: rest/routes/user.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+        )
+
+    if 'email' not in request.cookies:
+        status = 401
+        data = {'cookie': False}
+    else:
+        email = request.cookies.get('email')
+        user = User.query.filter_by(email=email).first()
+
+        if not user or not email == 'admin@admin.agh.edu.pl':
```
Comment (twrdyyy) [inline rest/routes/user.py:127]: do we have to hardcode that?

**Review by twrdyyy — CHANGES_REQUESTED** [review]

**Comment by twrdyyy** [comment]
@twistzzmc I will update .travis file after your changes so do not worry about checks

**Comment by twrdyyy** [comment]
@twistzzmc also please add docs to functions:
```
def func(arg1, arg2):
    """ 
    This function does nothing
    :param arg1: arg 1
    :param arg2: arg 2
    :returns: sth
```

this will clarify code

File: rest/routes/user.py (hunk 10 above)
Comment (twistzzmc) [reply rest/routes/user.py:127]: We need to hardcode some form of admin email style or put in is_admin field in database. I could change it to anything ending with "@admin.agh.edu.pl" is an admin. I don't see any other way

**Review by twrdyyy — APPROVED** [review]
LGTM <3

**Review by jmysliv — CHANGES_REQUESTED** [review]

**Review by jmysliv — APPROVED** [review]
LGTM <3

**Pull request merged**